README.dev.md

```diff
@@ -131,10 +115,9 @@ This section describes how to make a release in 3 parts:
 
 ### (1/3) Preparation
 
-1. Update the <CHANGELOG.md> (don't forget to update links at bottom of page)
-2. Verify that the information in [`CITATION.cff`](CITATION.cff) is correct.
-3. Make sure the [version has been updated](#versioning).
-4. Run the unit tests with `pytest -v`
+1. Verify that the information in [`CITATION.cff`](CITATION.cff) is correct.
```

Comment:
If we remove the sentence about the changelog here, we should make sure that they actually generate a changelog upon making the github release. Perhaps a somewhat more elaborate explanation on how to make a release in github would be  nice?